pkg/reloader/reloader.go

```diff
@@ -218,9 +218,15 @@ func (r *Reloader) Watch(ctx context.Context) error {
 		}
 
 		if err := r.apply(ctx); err != nil {
-			// Critical error.
-			// TODO(bwplotka): There is no need to get process down in this case and decrease availability, handle the error in different way.
-			return err
+			switch err := errors.Cause(err).(type) {
+			case *os.PathError:
+				r.watchErrors.Inc()
```

Comment:
To make these errors more visible, we can add labels to `watchErrors` and increment counter by the type.